src/main/scala/org/broadinstitute/dsde/firecloud/utils/PerformanceLogging.scala

```diff
@@ -1,13 +1,24 @@
 package org.broadinstitute.dsde.firecloud.utils
 
+import java.time.{Instant, LocalDateTime, ZoneId}
+import java.time.format.DateTimeFormatter
+
 import com.typesafe.scalalogging.Logger
 import org.slf4j.LoggerFactory
 
 trait PerformanceLogging {
 
   val perfLogger: Logger = Logger(LoggerFactory.getLogger("PerformanceLogging"))
+  val systemTimeZone = ZoneId.systemDefault()
+  val dateFormatter = DateTimeFormatter.ISO_LOCAL_TIME
+
+  def perfmsg(caller: String, msg: String, start: Instant, finish: Instant): String = {
+    lazy val elapsed = finish.toEpochMilli - start.toEpochMilli
+
+    lazy val startStr = LocalDateTime.ofInstant(start, systemTimeZone).format(dateFormatter)
+    lazy val finishStr = LocalDateTime.ofInstant(finish, systemTimeZone).format(dateFormatter)
 
-  def perfmsg(caller: String, time: Long) =              s"[$caller] took [$time] ms"
-  def perfmsg(caller: String, time: Long, msg: String) = s"[$caller] took [$time] ms: $msg"
+    s"[$caller] took [$elapsed] ms, start [$startStr], finish [$finishStr]: $msg"
```

Comment:
I like this convention of wrapping in brackets. Do you know where this came from?

Reply:
@dmohs I made it up (tho I've seen it in many other logs elsewhere). I figured that if we ever wanted to machine-parse these logs, it's a good compromise between human-readable and machine-parsable.